build/patches/Timezone-Override.patch

```diff
@@ -0,0 +1,1379 @@
++    }
++    public static void setCustomTimezone(BrowserContextHandle browserContextHandle, String timezone) {
++        WebsitePreferenceBridgeJni.get().setCustomTimezone(browserContextHandle, timezone);
++    }
++
+     @VisibleForTesting
+     @NativeMethods
+     public interface Natives {
+@@ -734,5 +743,7 @@ public class WebsitePreferenceBridge {
+         void setSensorsEnabled(BrowserContextHandle browserContextHandle, boolean enabled);
+         void setSoundEnabled(BrowserContextHandle browserContextHandle, boolean enabled);
+         void setVrEnabled(BrowserContextHandle browserContextHandle, boolean enabled);
++        String getCustomTimezone(BrowserContextHandle browserContextHandle);
++        void setCustomTimezone(BrowserContextHandle browserContextHandle, String timezone);
+     }
+ }
+diff --git a/components/browser_ui/site_settings/android/website_preference_bridge.cc b/components/browser_ui/site_settings/android/website_preference_bridge.cc
+--- a/components/browser_ui/site_settings/android/website_preference_bridge.cc
++++ b/components/browser_ui/site_settings/android/website_preference_bridge.cc
+@@ -1428,3 +1428,19 @@ static jboolean JNI_WebsitePreferenceBridge_GetMicManagedByCustodian(
+   return IsContentSettingManagedByCustodian(
+       jbrowser_context_handle, ContentSettingsType::MEDIASTREAM_MIC);
+ }
++
++static void JNI_WebsitePreferenceBridge_SetCustomTimezone(
++    JNIEnv* env,
++    const JavaParamRef<jobject>& jbrowser_context_handle,
++    const JavaParamRef<jstring>& timezone) { 
++  std::string new_timezone = ConvertJavaStringToUTF8(env, timezone);
++  GetHostContentSettingsMap(jbrowser_context_handle)->SetTimezoneOverrideValue(new_timezone);
++}
++
++static base::android::ScopedJavaLocalRef<jstring> JNI_WebsitePreferenceBridge_GetCustomTimezone(
++    JNIEnv* env,
++    const JavaParamRef<jobject>& jbrowser_context_handle) {
++  std::string timezone;
++  GetHostContentSettingsMap(jbrowser_context_handle)->GetTimezoneOverrideValue(timezone);
++  return ConvertUTF8ToJavaString(env, timezone);
++}
+diff --git a/components/browser_ui/strings/android/site_settings.grdp b/components/browser_ui/strings/android/site_settings.grdp
+--- a/components/browser_ui/strings/android/site_settings.grdp
++++ b/components/browser_ui/strings/android/site_settings.grdp
+@@ -66,6 +66,9 @@
+   <message name="IDS_VR_PERMISSION_TITLE" desc="Title of the permission to use Virtual Reality [CHAR-LIMIT=32]">
+     Virtual reality
+   </message>
++  <message name="IDS_TIMEZONE_OVERRIDE_PERMISSION_TITLE" desc="Title of the permission to use TimeZone Override [CHAR-LIMIT=32]">
++    Timezone override
++  </message>
+ 
+   <!-- Site settings global toggles -->
+ 
+@@ -468,6 +471,38 @@
+     Block sites from playing protected content
+   </message>
+ 
++  <!-- Timezone override (bromite #331) -->
++  <message name="IDS_WEBSITE_SETTINGS_TIMEOVERRIDE_INFO" desc="Primary text explaining system timezone feature.">
++    bla bla bla
```

Comment:
This needs to be filled in.

Reply:
yes, I leave it up to you to decide the text